src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs

```diff
@@ -206,12 +207,25 @@ private async Task CheckTokenAsync(CodeFixContext context, Document document, Sy
         private static async Task<string> GetInsertionTextAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
         {
             var service = CompletionService.GetService(document);
-            var change = await service.GetChangeAsync(document, item, commitCharacter: null, cancellationToken).ConfigureAwait(false);
-            var text = change.TextChange.NewText;
+            var provider = service.GetProvider(item, document.Project);
+
+            var text = await GetInsertionTextAsync().ConfigureAwait(false);
+
             var nonCharIndex = text.IndexOfAny(s_punctuation);
             return nonCharIndex > 0
                 ? text[0..nonCharIndex]
                 : text;
+
+            async Task<string> GetInsertionTextAsync()
+            {
+                // Snippets perform extremely complex logic to determine their text change.  We don't care about that,
+                // we only want to know what the text was that was shown the user (like `if`).
+                if (provider is AbstractSnippetCompletionProvider)
+                    return item.DisplayText;
```

Comment:
I wonder if we could simply do this for all, for any complex edit that differ from display test, spellchecker might not b eable to handle the change well anyway.

Reply:
WDYM :)  

Note: spell-checking does handle cases where the text is different i believe.  For example it can emit `@int` or `[integer]`.